pages/login/LoginHandler.inc.php

```diff
@@ -169,12 +180,71 @@ function requestResetPassword($args, $request) {
 		$this->setupTemplate($request);
 		$templateMgr = TemplateManager::getManager($request);
 
+		// Check if reCaptcha is enabled and validate it if so
+		$captchaEnabled = Config::getVar('captcha', 'captcha_on_password_reset') && Config::getVar('captcha', 'recaptcha');
+		if ($captchaEnabled) {
+			import('lib.pkp.classes.form.Form');
+			$form = new Form('');
+			$form->setData('g-recaptcha-response', $request->getUserVar('g-recaptcha-response'));
+
+			import('lib.pkp.classes.form.validation.FormValidatorReCaptcha');
+			$captchaValidator = new FormValidatorReCaptcha(
+				$form,
+				$request->getRemoteAddr(),
+				'common.captcha.error.invalid-input-response',
+				$request->getServerHost()
+			);
+
+			if (!$captchaValidator->isValid()) {
+				$templateMgr->assign([
+					'captchaEnabled' => true,
+					'reCaptchaHtml' => '<div class="g-recaptcha" data-sitekey="' . Config::getVar('captcha', 'recaptcha_public_key') . '"></div><label for="g-recaptcha-response" style="display:none;">Recaptcha response</label>',
+					'error' => $captchaValidator->getMessage(),
+					'email' => $request->getUserVar('email'),
+				]);
+				$templateMgr->display('frontend/pages/userLostPassword.tpl');
+				return;
+			}
+		}
+
 		$email = $request->getUserVar('email');
 		$userDao = DAORegistry::getDAO('UserDAO'); /** @var UserDAO $userDao */
 		$user = $userDao->getUserByEmail($email);
 
+		$rateLimitingEnabled = true;
```

Comment:
I think this should be added to a configuration, and perhaps disabled by default to avoid breaking changes.
Then a user can opt-in for both the recaptcha and the rate limiter.